pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/columnminmaxvalue/ColumnMinMaxValueGenerator.java

```diff
@@ -124,227 +131,251 @@ private List<String> getColumnsToAddMinMaxValue() {
+              for (float value : values) {
                 min = Math.min(min, value);
                 max = Math.max(max, value);
               }
-            } else {
-              for (int docId = 0; docId < numDocs; docId++) {
-                float[] values = rawIndexReader.getFloatMV(docId, readerContext);
-                for (float value : values) {
-                  min = Math.min(min, value);
-                  max = Math.max(max, value);
-                }
-              }
             }
-            SegmentColumnarIndexCreator.addColumnMinMaxValueInfo(_segmentProperties, columnName, String.valueOf(min),
-                String.valueOf(max), storedType);
-            break;
           }
-          case DOUBLE: {
-            double min = Double.POSITIVE_INFINITY;
-            double max = Double.NEGATIVE_INFINITY;
-            if (isSingleValue) {
-              for (int docId = 0; docId < numDocs; docId++) {
-                double value = rawIndexReader.getDouble(docId, readerContext);
+          minValue = min;
+          maxValue = max;
+          break;
+        }
+        case DOUBLE: {
+          double min = Double.POSITIVE_INFINITY;
+          double max = Double.NEGATIVE_INFINITY;
+          if (isSingleValue) {
+            for (int docId = 0; docId < numDocs; docId++) {
+              double value = rawIndexReader.getDouble(docId, readerContext);
+              min = Math.min(min, value);
+              max = Math.max(max, value);
+            }
+          } else {
+            for (int docId = 0; docId < numDocs; docId++) {
+              double[] values = rawIndexReader.getDoubleMV(docId, readerContext);
+              for (double value : values) {
                 min = Math.min(min, value);
                 max = Math.max(max, value);
               }
-            } else {
-              for (int docId = 0; docId < numDocs; docId++) {
-                double[] values = rawIndexReader.getDoubleMV(docId, readerContext);
-                for (double value : values) {
-                  min = Math.min(min, value);
-                  max = Math.max(max, value);
-                }
-              }
             }
-            SegmentColumnarIndexCreator.addColumnMinMaxValueInfo(_segmentProperties, columnName, String.valueOf(min),
-                String.valueOf(max), storedType);
-            break;
           }
-          case STRING: {
-            String min = null;
-            String max = null;
-            if (isSingleValue) {
-              for (int docId = 0; docId < numDocs; docId++) {
-                String value = rawIndexReader.getString(docId, readerContext);
+          minValue = min;
+          maxValue = max;
+          break;
+        }
+        case BIG_DECIMAL: {
+          Preconditions.checkState(isSingleValue, "Unsupported multi-value BIG_DECIMAL column: %s", columnName);
+          BigDecimal min = null;
+          BigDecimal max = null;
+          for (int docId = 1; docId < numDocs; docId++) {
```

Comment:
why is this starting at `docId` 1 instead of 0?

Reply:
Good catch! This is a bug (initially I put first value as min/max, then forgot to change it back..)